lib/Doctrine/Annotations/Metadata/AnnotationMetadata.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata;
+
+use Doctrine\Annotations\Metadata\Exception\TooManyDefaultProperties;
+use function array_combine;
+use function array_filter;
+use function array_map;
+use function array_values;
+use function count;
+
+/**
+ * @internal
```

Comment:
Let's add a line to explain:

 * why this is internal
 * what this metadata represents (it's unclear to the reader, but this is about an annotation's definition, rather than an annotation instance)

Reply:
I decided to remove `@internal` from AnnotationMetadata, PropertyMetadata and AnnotationTarget because since MetadataCollection is an extension point, it should not depend on internal elements.